apps/sync/js/crypto/fxsyncwebcrypto.js

```diff
@@ -210,8 +210,7 @@ ${collectionName}`);
                                    payload.ciphertext)
           .then(recordArrayBuffer => {
         var recordObj;
-        const recordJSON = String.fromCharCode.apply(
-            null,
+        const recordJSON = StringConversion.byteArrayToUtf16String(
```

Comment:
As discussed on irc, since the SyncStorage service always encodes BSOs in UTF8 (see https://docs.services.mozilla.com/storage/apis-1.5.html#basic-storage-object) we should not interpret them as UTF16.